### ISSUE TITLE
More details in CheckEnvironment.php

### ISSUE THREAD

**Issue by amosfolz** [userfrosting/UserFrosting#934]
If the folders `/log` `/cache` or `/session` have been accidentally deleted or moved, the error does not give helpful information. 

![clipboard - february 8 2019 1-55 pm](https://user-images.githubusercontent.com/33728190/52534957-4e6ff000-2d16-11e9-89bc-5c9cfab1571f.png)

Per Alex Weissman, "this ought to display the full path".

https://github.com/userfrosting/UserFrosting/blob/4dda14f611b6de5764e05d22538f5c179b0a18f7/app/sprinkles/core/src/Util/CheckEnvironment.php#L266-L274

**Comment by lcharette** [comment]
Related #630.

**Comment by amosfolz** [comment]
I understand this is part of some larger changes that are planned but I thought just changing the error message could help with this specific issue for now: 
https://github.com/amosfolz/UserFrosting/commit/6870f7a88cfdaad3cb266be49675f0a084c45d49

**Comment by lcharette** [comment]
Note because of the `foreach` above, this has the potential of displaying the same message more than once. That's why the ` $file`  variable is used. The real fix would be to see why it contains `0` instead of the file name. I guess it has something to do with the locator returning an empty array.

**Comment by amosfolz** [comment]
Ahh you bring up some good points. 

It seems the locator returns `0` if it can not find anything. Here is the result of renaming/deleting the `sessions` directory.
```
[2019-03-17 19:00:30] debug.DEBUG: shouldBeWriteable [] []
[2019-03-17 19:00:30] debug.DEBUG: Array
(
    [/home/vagrant/userfrosting/app/logs] => 1
    [/home/vagrant/userfrosting/app/cache] => 1
    [0] => 1
)
```

**Comment by amosfolz** [comment]
Would it be appropriate to add a new function that checks for existence prior to running the `checkPermissions()`? It could be something like this:
```
        $shouldExist = [
        "log"     => $this->locator->findResource('log://'),
        "cache"   => $this->locator->findResource('cache://'),
        "session" => $this->locator->findResource('session://')
        ];

        foreach ($shouldExist as $directory => $path) {
          if ($path == null) {
            //Show error message for missing directory here. "$directory is missing..."
          }
        }
```

**Comment by lcharette** [comment]
Those two different array are making thing complicated...

https://github.com/amosfolz/UserFrosting/blob/6870f7a88cfdaad3cb266be49675f0a084c45d49/app/sprinkles/core/src/Util/CheckEnvironment.php#L267-L271

https://github.com/amosfolz/UserFrosting/blob/6870f7a88cfdaad3cb266be49675f0a084c45d49/app/sprinkles/core/src/Util/CheckEnvironment.php#L275-L278

**Comment by lcharette** [comment]
Fixed in PR #958 and released as of 4.2.1
